plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -1538,4 +1612,66 @@ protected function get_product_data( $product, $context = 'view' ) {
 
 		return $data;
 	}
+
+	/**
+	 * Get the related products.
+	 * Related products is handled by the wc_get_related_products() core function.
+	 * Currently, the data that define the related products are the categories and tags.
+	 * This endpoint accepts and combines the categories and tags parameters.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return object
+	 */
+	public function get_related_items( $request ) {
+		$id         = $request->get_param( 'id' );
+		$categories = $request->get_param( 'related_categories' );
+		$tags       = $request->get_param( 'related_tags' );
+		$combine    = $request->get_param( 'combine' );
+		$attributes = $request->get_param( 'attributes' );
+
+		/*
+		 * If the `categories` params is defined,
+		 * filter the categories used by the
+		 * wc_get_related_products() function.
+		 */
+		if ( ! empty( $categories ) ) {
+			add_filter(
+				'woocommerce_get_related_product_cat_terms',
+				function( $product_categories ) use ( $categories, $combine ) {
+					if ( ! $combine ) {
+						return $categories;
+					}
+
+					return array_unique( array_merge( $product_categories, $categories ) );
+				}
+			);
+		}
+
+		/*
+		 * If the `tags` params is defined,
+		 * filter the tags used by the
+		 * wc_get_related_products() function.
+		 */
+		if ( ! empty( $tags ) ) {
+			add_filter(
+				'woocommerce_get_related_product_tag_terms',
+				function( $product_tags ) use ( $tags, $combine ) {
+					if ( ! $combine ) {
+						return $tags;
+					}
+
+					return array_unique( array_merge( $product_tags, $tags ) );
+				}
+			);
+		}
+
+		$this->related_producs_ids = wc_get_related_products( $id );
```

Comment:
I noticed that the `wc_get_related_products` actually caches the query data using a transient, based on the product id, the limit, and excluded id's. This does not work very well for us, given that as soon as a user triggers the related products and then updates tags or categories, the changes are not reflected.
This will require us to move the logic from `wc_get_related_products` in here without the use of the `transient` or still use a `transient` but including tags & categories as part of the key.